### PR TITLE
Fix related posts

### PR DESCRIPTION
**Pull request by kurtw**
Phew. This one was a bit of a doozy. This does a few things but is all related to getting this related posts mechanism working once and for all. Basically, it's adding a script to update the ES index with new settings (something we didn't have) then updating the tags field in the mapping so we can perform lowercase searches for content while still maintain the original case in places like the filters, then it adds a new search method on the Query class in sheerlike to actually get the posts by the lowercase tag.

Then another commit updates the templates to actually render things properly, adding a generic streamfield template.

## Additions

- New script to update es settings
- New method to find documents based on tags
- New newsroom query
- New analyzer for tags
- Template for generic rendering of sidefoot streamfield

## Changes

- All page templates now use this generic sidefoot streamfield template

## Testing

- First, run `./cfgov/manage.py runscript update_es_settings`
- Then, run `./cfgov/manage.py sheer_index -r`
- Go [here](http://content.localhost:8000/about-us/payments-harmed-consumers/civil-penalty-fund/consumer-education-financial-literacy/). If you can see related posts, then hallelujah.
- Check /blogs/ and /newsroom/ to make sure those still work

## Review

- @richaagarwal 
- @kave 
- @rosskarchner

### CODE REVIEW

**Comment by richaagarwal** [comment]
I get this error back when trying to run `./cfgov/manage.py runscript sheer_index -r`:
```Usage: ./cfgov/manage.py runscript [options] script [script ...]

Runs a script in django context.

./cfgov/manage.py: error: no such option: -r ```

File: cfgov/jinja2/v1/_includes/templates/streamfield-sidefoot.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% import 'molecules/related-posts.html' as related_posts with context %}
+{% import 'molecules/related-metadata.html' as related_metadata with context %}
+
+
+{% macro render(streamfield, half_width=false) %}
+    {% for block in streamfield %}
+        {% if 'related_posts' in block.block_type %}
```
Comment (kave) [inline cfgov/jinja2/v1/_includes/templates/streamfield-sidefoot.html]: i think we need to add the div 
```
<div class="block block__flush-top">
```
in here for spacing issues

File: cfgov/jinja2/v1/landing-page/index.html [hunk 2]
```diff
@@ -41,17 +41,5 @@
 {%- endblock %}
 
 {% block content_sidebar scoped -%}
-    {% for block in page.sidefoot %}
-        <div class="block block__flush-top">
-            {% if 'related_posts' in block.block_type %}
-                {{ related_posts.render(block) }}
-            {% elif 'heading' in block.block_type %}
-                <h3 class="o-sidebar-breakout_heading">
```
Comment (kave) [inline cfgov/jinja2/v1/landing-page/index.html]: is this conditional no longer necessary?

**Comment by kave** [comment]
@richaagarwal try `./cfgov/manage.py sheer_index -r`without the `runscript` that was a typo

**Comment by kave** [comment]
@kurtw can you document somewhere so we remember to do these steps not only on flapjack/refresh but also on beta/build `./cfgov/manage.py runscript update_es_settings`?

File: cfgov/jinja2/v1/landing-page/index.html (hunk 2 above)
Comment (kurtw) [inline cfgov/jinja2/v1/landing-page/index.html]: `heading` doesn't exist in the sidefoot streamfield

Comment (kave) [inline cfgov/jinja2/v1/landing-page/index.html]: cool

**Comment by kave** [comment]
besides the template questions. works as stated :+1:

File: cfgov/jinja2/v1/_includes/templates/streamfield-sidefoot.html (hunk 1 above)
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/templates/streamfield-sidefoot.html]: resolved.

**Pull request merged**